lib/datasource/deb/index.ts

```diff
@@ -0,0 +1,306 @@
+        return extractedFile;
+      } catch (e) {
+        logger.warn(
+          "Couldn't download package file with compression " +
+            compression +
+            ' from ' +
+            basePackageUrl
+        );
+      }
+    }
+
+    throw 'No compression standard worked for ' + basePackageUrl;
+  }
+
+  /**
+   *
+   * @param cfg
+   * @param basePackageUrl
+   * @param compression
+   * @returns a boolean indicating if the file was modified
+   */
+  async downloadPackageFile(
+    basePackageUrl: string,
+    compression: string,
+    compressedFile: string
+  ): Promise<boolean> {
+    let lastTimestamp: Date = null;
+    try {
+      const stats = await fs.stat(compressedFile);
+      lastTimestamp = stats.mtime;
+    } catch (e) {
+      // ignore if the file doesnt exist
+    }
+
+    const packageUri = basePackageUrl + '/Packages.' + compression;
+    logger.debug(
+      'Downloading package file from ' + packageUri + ' as ' + compressedFile
+    );
+    let needsToDownload = true;
+    // we can use If-Modified-Since to avoid that we are redownloaded the file
+    // More information can be found here: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/If-Modified-Since
+    if (lastTimestamp !== null) {
+      const downloadOptions: HttpOptions = {};
+      downloadOptions.headers = {
+        'If-Modified-Since': lastTimestamp.toUTCString(),
+      };
+      const headResult = await this.http.head(packageUri, downloadOptions);
+      needsToDownload = headResult.statusCode !== 304;
+    }
+
+    if (needsToDownload) {
+      const readStream = this.http.stream(packageUri);
+      const writeStream = fs.createWriteStream(compressedFile);
+
+      await fs.pipeline(readStream, writeStream);
+    }
+
+    return needsToDownload;
+  }
+
+  async probeExtractedPackage(
+    extractedFile,
+    packageName
+  ): Promise<ReleaseResult | null> {
+    // read line by line to avoid high memory consumption as the extracted Packages
+    // files can be multiple MBs in size
+    const rl = readline.createInterface({
+      input: fs.createReadStream(extractedFile),
+      terminal: false,
+    });
+    let pd: PackageDescription = {};
+    for await (const line of rl) {
+      if (line === '') {
+        // now we should have all information available
+        if (pd.Package === packageName) {
+          return { releases: [{ version: pd.Version }], homepage: pd.Homepage };
+        }
+        pd = {};
+        continue;
+      }
+
+      for (let i = 0; i < this.requiredPackageKeys.length; i++) {
+        if (line.startsWith(this.requiredPackageKeys[i])) {
+          pd[this.requiredPackageKeys[i]] = line
+            .substring(this.requiredPackageKeys[i].length + 1)
+            .trim();
+          break;
+        }
+      }
+    }
+
+    // if the searched package is the last one, this is required
+    if (pd.Package === packageName) {
+      return { releases: [{ version: pd.Version }], homepage: pd.Homepage };
```

Comment:
Are you sure we only ever have one version?

Reply:
So yes and no, every Packages file must only have a package once but a package might be specified in multiple repositories which I've added now.